pkg/kubectl/cmd/get/get.go

```diff
@@ -547,13 +546,14 @@ func (o *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) e
 			w.Flush()
 			w.SetRememberedWidths(nil)
 
-			// add linebreak between resource groups (if there is more than one)
-			// skip linebreak above first resource group
-			if lastMapping != nil && !o.NoHeaders {
-				// If we've written output since the last time we started a new set of headers, write an empty line to separate object types
-				if written != trackingWriter.Written {
-					fmt.Fprintln(w)
-					written = trackingWriter.Written
+			// add linebreaks between resource groups (if there is more than one)
+			// when it satisfies all following 3 conditions:
+			// 1) it's not the first resource group
+			// 2) it has row header
+			// 3) we've written output since the last time we started a new set of headers
+			if lastMapping != nil && !o.NoHeaders && trackingWriter.Written > 0 {
+				if trackingWriter.Written > 0 {
```

Comment:
no need to check `trackingWriter.Written > 0` twice

Reply:
Forgot to remove the original line when I moved it up.. Updated.